src/wagtail_vector_index/consumers.py

```diff
@@ -0,0 +1,101 @@
+import asyncio
+import logging
+
+from channels.generic.http import AsyncHttpConsumer
+from django import forms
+from django.apps import apps
+from django.http import QueryDict
+
+logger = logging.Logger(__name__)
+
+
+class WagtailVectorIndexQueryParamsForm(forms.Form):
+    """Provides a form for validating query parameters."""
+
+    query = forms.CharField(max_length=255, required=True)
+    page_type = forms.CharField(max_length=255, required=True)
+
+
+class WagtailVectorIndexSSEConsumer(AsyncHttpConsumer):
+    """
+    A Django Channels consumer for handling Server-Sent Events (SSE) related to WagtailVectorIndex queries.
+
+    Methods:
+        handle: The main entry point for processing HTTP requests, including SSE connections.
+        process_prompt: Processes the incoming prompt and sends SSE updates.
+
+    Note:
+        This consumer expects the following query parameters in the URL:
+        - 'query': The search query.
+        - 'page_type': The type of Wagtail page to search.
```

Comment:
**Suggestion:** Vector indexes may not to be related to Wagtail pages - they could be Documents, other Django models, or some completely independent data source. If we can take the index name here and look it up in the registry, that would enable this endpoint to work across all defined indexes.

Reply:
Had a look into this, however, could do with some pointers as I can't figure out why `self.object_type.bulk_from_documents` in `aquery` errors.